graphql/query_resolver.go

```diff
@@ -822,8 +843,8 @@ func (r *queryResolver) MainlineCommits(ctx context.Context, options MainlineCom
 	matchingVersionCount := 0
 
 	// We only want to return the PrevPageOrderNumber if a user is not on the first page
-	if options.SkipOrderNumber != nil {
-		prevPageCommit, err := model.GetPreviousPageCommitOrderNumber(ctx, projectId, utility.FromIntPtr(options.SkipOrderNumber), limit, requesters)
+	if skipOrderNumber != 0 {
```

Comment:
Is setting `options.SkipOrderNumber` intentionally at zero a valid case? Seems that this conditional would evaluate true before this change, just making sure that we do want to make a change to skip that case

Reply:
Yes, the if-statements using a different check was intentional. The new logic means that the revision only gets used when `options.skipOrderNumber == nil`. It’s done this way because the way to paginate commits is through the skipOrderNumber, and so `options.revision` and `options.skipOrderNumber` need to share that variable with each other.

So with this new logic, even if `options.skipOrderNumber` is `nil`, there could be previous pages if we've jumped to a particular commit, and we'd want to calculate the previous page offset. I hope that made sense 😅